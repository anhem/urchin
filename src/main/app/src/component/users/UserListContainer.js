import React, {Component} from 'react'
import {connect} from 'react-redux'
import {getUsers} from '../../action/userAction'
import UserList from "./UserList";

export class UserListContainer extends Component {

    componentWillMount() {
        this.props.getUsers();
    }

    render() {
        return (
            <UserList
                users={this.props.users}
            />
        )
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        getUsers: () => {
            dispatch(getUsers())
        }
    }
};

const mapStateToProps = (state) => {
    return {
        users: state.userData.users || []
    }
};

export default connect(mapStateToProps, mapDispatchToProps)(UserListContainer)