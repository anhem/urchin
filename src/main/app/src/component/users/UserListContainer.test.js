import React from 'react';
import {mount, shallow} from 'enzyme'
import toJson from 'enzyme-to-json';
import {UserListContainer} from "./UserListContainer";

describe('UserListContainer', () => {

    let props = {
        getUsers: jest.fn()
    };

    it('renders without crashing', () => {
        expect(mount(<UserListContainer {...props}/>).length).toEqual(1);
    });

    it('match snapshot', () => {
        expect(toJson(shallow(<UserListContainer {...props}/>))).toMatchSnapshot();
    });
});
