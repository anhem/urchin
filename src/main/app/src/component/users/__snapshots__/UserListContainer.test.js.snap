// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListContainer match snapshot 1`] = `<UserList />`;
